File: booleanOperators.py
test1 = True

test2 = False

#* --------------------------------- not gates -------------------------------- #

def notGate(a):
    return True if not a else False

#* ---------------------------------- or gate --------------------------------- #

def orGate(a, b):
    return True if a or b else False

#* --------------------------------- and gate --------------------------------- #

def andGate(a, b):
    return True if a and b else False

#* --------------------------------- xor gate --------------------------------- #

def xorGate(a, b):
    return True if a != b else False

print(notGate(test1))
print(orGate(test1, test2))
print(andGate(test1, test2))
print(xorGate(test1, test2))

File: booleanOperators.js
//~ Learn/Understand how various boolean operators and 
//~ logic gates function.

const test1 = true;

const test2 = false;

//* -------------------------------- not gates ------------------------------- */

function notGate(a) {
    return !a ? true : false;
}

//* -------------------------------- or gates -------------------------------- */

function orGate(a, b) {
    return (a || b) ? true : false;
}

//* -------------------------------- and gate -------------------------------- */

function andGate(a, b) {
    return (a && b) ? true : false;
}

//* -------------------------------- xor gate -------------------------------- */

function xorGate(a, b) {
    return (!a != !b) ? true : false;
}

console.log(notGate(test1));
console.log(orGate(test1, test2));
console.log(andGate(test1, test2));
console.log(xorGate(test1, test2));
